### PR TITLE
feat(server): opt-in audience subscribing on registration

### PR DESCRIPTION
**Pull request by didimitrie**
Adds a newsletter registration consent box on the register screen.

### CODE REVIEW

File: utils/helm/speckle-server/templates/server/deployment.yml [hunk 1]
```diff
@@ -276,6 +276,25 @@ spec:
           - name: EMAIL_FROM
             value: "{{ .Values.server.email.from }}"
           {{- end }}
+          
+          # *** Mailchimp ***
```
Comment (didimitrie) [inline utils/helm/speckle-server/templates/server/deployment.yml:280]: cc @iainsproat: this is where your magic will be needed. PS: we have these values in the shared vault.

File: packages/frontend-2/components/auth/RegisterWithEmailBlock.vue [hunk 2]
```diff
@@ -43,6 +43,23 @@
...
     />
     <FormButton submit full-width class="mt-4" :disabled="loading">Sign up</FormButton>
+    <div
+      class="mt-3 text-xs flex items-center justify-center text-foreground-2 space-x-2"
+    >
+      <!-- 
+        Note the newsletter consent box is here because i got very confused re layout of the panel
+        and didn't figure out a better way to put it where i needed it to be
+       -->
+      <label for="newsletter" class="sxxr-only">
```
Comment (fabis94) [inline packages/frontend-2/components/auth/RegisterWithEmailBlock.vue:53]: Why not use `<FormInputCheckbox/>`?

File: packages/server/modules/shared/helpers/envHelper.ts [hunk 3]
```diff
@@ -81,6 +81,22 @@ export function getOidcName() {
   return process.env.OIDC_NAME
 }
 
+export function getMailchimpConfig() {
+  if (
+    !process.env.MAILCHIMP_API_KEY ||
```
Comment (fabis94) [inline packages/server/modules/shared/helpers/envHelper.ts:86]: How will this work locally? Should this just be disabled or should we set up actual mailchimp lists for dev purposes?

**Review by fabis94 — CHANGES_REQUESTED** [review]

File: packages/frontend-2/components/auth/RegisterPanel.vue [hunk 4]
```diff
@@ -63,6 +63,10 @@ graphql(`
   }
 `)
 
+const newsletterConsent = ref(false)
+
+provide('newsletterconsent', newsletterConsent)
```
[the remark this answers is not in the document]
Comment (didimitrie) [reply packages/frontend-2/components/auth/RegisterPanel.vue:68]: It's tricky - i need the checkbox to be visually in the register with email block, but that var needs to be available in the third party call; this was an easier way to declare it up and inject it where in needed it without mutating props and stuff. I felt weird passing in a ref via props...

File: packages/frontend-2/components/auth/RegisterWithEmailBlock.vue (hunk 2 above)
Comment (didimitrie) [reply packages/frontend-2/components/auth/RegisterWithEmailBlock.vue:53]: i needed to style it a bit more and didn't yet know how to publish a new version of the ui components.

File: packages/server/modules/shared/helpers/envHelper.ts (hunk 3 above)
Comment (didimitrie) [reply packages/server/modules/shared/helpers/envHelper.ts:86]: added an extra env var, we can ignore it locally

File: packages/server/modules/auth/services/mailchimp.ts [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+      email_address: user.email,
+      merge_fields: {
+        EMAIL: user.email,
+        FNAME: first,
+        LNAME: second,
+        FULLNAME: user.name // NOTE: this field needs to be set in the audience merge fields
+      }
+    })
+  } catch (e) {
+    logger.warn(e, 'Failed to register user to newsletter.')
```
[the remark this answers is not in the document]
Comment (didimitrie) [reply packages/server/modules/auth/services/mailchimp.ts:42]: extra enabled env var added helps resolve this - we'll log only when the newsletter is explicitly on and something else goes wrong.

**Pull request merged**